### PR TITLE
course: Advanced RAG implementation

### PR DESCRIPTION
**Pull request by parambharat**
This PR contains the course work and related code for the RAG-IRL course.

### CODE REVIEW

**Comment by ayulockin** [comment]
@parambharat shouldn't we move the `data` dir to an artifact or as weave dataset?

File: rag-advanced/README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Advanced RAG IRL
+
+This repository contains the code for the Advanced RAG IRL course.
```
Comment (agata-mlynarczyk) [inline rag-advanced/README.md:3]: could you add a link to the course? https://www.wandb.courses/courses/rag-in-production

Comment (agata-mlynarczyk) [reply rag-advanced/README.md:3]: before launching the course we should also add a quick overview, course learning objectives etc. but we can do so afterwards

**Comment by ayulockin** [comment]
nit: how about we use the naming of the notebooks like - `00_name.ipynb`, `01_name1.ipynb`, etc?

**Comment by ayulockin** [comment]
The data that we are downloading contains only `.md` file format?

File: rag-advanced/notebooks/scripts/retriever.py [hunk 2]
```diff
@@ -0,0 +1,162 @@
+import bm25s
+import Stemmer
+import weave
+from scipy.spatial.distance import cdist
+from sklearn.feature_extraction.text import TfidfVectorizer
+from .embedding import sync_embed
+from .reranker import CohereReranker, FusionRanker
```
Comment (ayulockin) [inline rag-advanced/notebooks/scripts/retriever.py:7]: I don't see `reranker.py` file.

**Review by agata-mlynarczyk — APPROVED** [review]

**Pull request merged**